docs/guides/1238-split-plan-with-packages.md

```diff
@@ -0,0 +1,247 @@
+---
+slug: /1238/split-plan-with-packages
+displayed_sidebar: "0.2"
+---
+
+# Split a Dagger plan using packages
+
+Reading how to [split a plan with module](./1237-split-plan-with-module.md) is a prerequisite, as we will base the article on the project previously built.
+
+## Theory
+
+After relying on modules, let's explore local packages as an alternative way of splitting plans.
+
+After initializing and updating the project, a `cue.mod` folder should appear:
+
+```console
+$ tree  -L 2 cue.mod
+cue.mod
+├── module.cue
+├── pkg
+│   ├── dagger.io
+│   └── universe.dagger.io
+└── usr
+
+4 directories, 1 file
+```
+
+Let's make a more in-depth analysis of the folder structure. We already saw that `module.cue` could be used to split plans within a module. Let's explore the `pkg` and `usr` folders.
+
+- The `pkg` folder shouldn't be manually modified, apart from deleting one of its subfolder (for now). It contains the vendored packages retrieved from the `dagger project update` command.
+
+- The `usr` folder, on the contrary, is the place where you will be able to write your own packages. No vendoring program shall touch this folder, avoiding automatic mistakes.
+
+### Sructure of a package
+
+When importing packages in a Cue file, the folder structure in `cue.mod/usr` shall follow the import path:
+
+*Given the below Dagger plan importing a package and located at the root of your project:*
+
+```cue title="main.cue"
+package main
+
+import(
+    "github.com/foo/bar" // <= package imported, example not working yet
+)
+```
+
+*The package imported right above needs to follow the below structure in the `cue.mod/usr` folder:*
+
+```console
+$ tree cue.mod/usr
+cue.mod/usr
+└── github.com
+    └── foo
+        └── bar
+            └── foo.cue // name of file not important
+
+3 directories, 1 file
+```
+
+#### Inside the imported package
+
+```cue title="cue.mod/usr/github.com/foo/bar/foo.cue"
+package baz
+
+#test: "ok"
+```
+
+### Importing packages
+
+In the above example, our `foo.cue` file has a package name (`baz`) different than his parent directory (`bar`).
+
+This is done on purpose, as this is the main difference with modules: a module can only contain files whose package names are equal to name of their parent directories. This obligation doesn't apply to packages.
```

Comment:
```suggestion
This is done on purpose, as this is the main difference with modules: a module can only contain files whose package names are equal to the name of their parent directories. This obligation doesn't apply to packages.
```

Reply:
Some users are complaining about the fact that Cue's doc doesn't explain Cue concepts with actual examples. Here, we are doing exactly that: explain some Cue concepts in a more pragmatic way.

I understand the fragmentation issue, but why not let the users decide ? We can do exactly the same thing, and Cue will always support both.

I don't have a strong opinion on the subject, tbh